### PR TITLE
Release tasks

### PR DESCRIPTION
**Pull request by ruiqima**
## Description

PR to to complete pre-release checks. Removed unused imports, update dependencies versions, did code improvements, etc.

## Links and Data

#3737

## What To Review

All unused imports are removed, dependencies versions are aligned, necessary code improvement is done, version name is updated.

## How to Test

Build and run the sample viewer.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,21 +1,21 @@
 [versions]
 # ArcGIS Maps SDK for Kotlin version
-arcgisMapsKotlinVersion = "200.4.0-4189"
+arcgisMapsKotlinVersion = "200.4.0"
 # ArcGIS Maps SDK for Kotlin Toolkit version
-arcgisToolkitVersion = "200.4.0-4189"
+arcgisToolkitVersion = "200.4.0"
```
Comment (shubham7109) [inline gradle/libs.versions.toml:5]: I am hesitant to merge these changes into `v.next` as of now, or at least not until the day of release. Although I would like to merge the lint code improvements to `v.next`, removing the build tags and olympus here will break daily builds. 

Instead, we can prep a `release/200.4.0` branch with these final release changes which will eventually merge to `main`, like we have in previous releases. ([200.2](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/131),[200.3](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/141)). This way we can keep building the SV and update the release branch with any `v.next` improvments until day of release. 

For now, could you add in the build tag and olympus code in this PR to continue with PRs the lint code enhancements?

**Review by shubham7109 — COMMENTED** [review]
Nice quality code improvements here @ruiqima. Nice to revisit these code changes and improve the quality of the samples project once every release. 

I have one concern which I address in the comment below:

**Comment by shubham7109** [comment]
Added in a PR for release to `main`: https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/192

File: gradle/libs.versions.toml (hunk 1 above)
Comment (ruiqima) [reply gradle/libs.versions.toml:5]: Makes sense. Will add them back. Should I create a new release  branch now, or we can wait until the release note is ready?

Comment (ruiqima) [reply gradle/libs.versions.toml]: Just saw your second comment about release branch, nvm:)

**Comment by shubham7109** [comment]
@ruiqima I merged in the geoview scene samples, could you see if there if there are any lint improvements needed?

Also, might need to update the base branch. Thanks!

**Comment by ruiqima** [comment]
Updated with `v.next`. Re-ran inspect code, no lint improvement shows up.

**Review by shubham7109 — APPROVED** [review]

**Pull request merged**